src/main/java/com/nike/cerberus/service/AuthenticationService.java

```diff
@@ -402,6 +402,27 @@ private VaultAuthResponse generateToken(final String username, final Set<String>
     }
 
     /**
+     * Build a set of policies that is allowed for the specific IAM principal (e.g. an assumed user, or instance profile)
```

Comment:
This comment doesn't explain this very well.

    Builds the complete set of policies for an IAM principal by ensuring assumed roles and instance profiles are also bound to there respective roles. 

    ex: An instance profile arn arn:aws:iam::1111111111:instance-profile/example or an assumed role arn:aws:sts::1111111111:assumed-role/example/i-abcdef will get policies that were set up specifically for the authenticating arn and it will also get the policies for the base role arn:aws:iam::1111111111:role/example.

    This is done to copy the way IAM policies work in the AWS ecosystem.